### PR TITLE
[Button] Prevents disclosure icon from overflowing its container

### PR DESCRIPTION
**Pull request by mateus**
### WHY are these changes introduced?

The Button with disclosure icon is overflowing the content's container of the Button component. The padding should be 16px on the sides by the looks of it. 

Screen Shot 2022-06-28 at 3 07 47 PM

### WHAT is this pull request doing?

Updates the spacing the icon to make sure it stays within the boundaries of its container. 

Screen Shot 2022-06-28 at 2 37 43 PM

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useState} from 'react';
import {Page} from '../src';

export function Playground() {
    const [expanded, setExpanded] = useState(false);

  return (
    <Page title="Playground">
      <Button
        fullWidth
        textAlign="left"
        disclosure={expanded ? "up" : "down"}
        onClick={() => setExpanded(!expanded)}
      >
        {expanded ? "Show less" : "Show more"}
      </Button>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Button/Button.scss [hunk 1]
```diff
@@ -105,9 +105,8 @@
 
   &:last-child {
     // This compensates for the disclosure icon, which is substantially
-    // inset within the viewbox (and makes it look like there is too much
-    // spacing on the right of the button)
-    margin-right: calc(-1 * (var(--p-space-2)));
+    // inset within the viewbox
+    margin-right: calc(-1 * (var(--p-space-1)) - 2px);
```
Comment (alex-page) [inline polaris-react/src/components/Button/Button.scss]: What is the pixel value here? `2px`?

Comment (mateus) [reply polaris-react/src/components/Button/Button.scss]: It's a magic number to account for that extra spacing from the icon. Here's the button without the 2px subtraction 

Screen Shot 2022-06-28 at 3 18 24 PM

Comment (kyledurand) [reply polaris-react/src/components/Button/Button.scss]: Can we just use `--p-space-05`?

https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/spacing.ts#L8-L10

Comment (mateus) [reply polaris-react/src/components/Button/Button.scss]: Oh, absolutely. That will do it 👍🏻

Comment (mateus) [reply polaris-react/src/components/Button/Button.scss]: Actually, no. What it needs is -6px of margin right. Should I just hardcode that value instead?

Comment (mateus) [reply polaris-react/src/components/Button/Button.scss]: Added a custom property for it called `--pc-button-disclosure-icon-offset`

**Review by kyledurand — APPROVED** [review]
You could also do this but I'm happy using -6px directly here
`calc(-1 * var(--p-space-1) - var(--p-space-05));`

**Review by alex-page — APPROVED** [review]

**Pull request merged**